drgn/commands/_builtin/crash/vm.py

```diff
@@ -0,0 +1,179 @@
+            vm_end = vma.vm_end
+            vm_flags = vma.vm_flags
+
+            # Get file path
+            file_path = vma_name(vma, mm)
+"""
+    )
+    code.print()
+
+
+def _print_task_vm_info(prog: Program, task: Object) -> None:
+    """Print virtual memory info for a single task"""
+    print_task_header(task)
+
+    mm = task.mm.read_()
+    if not mm:
+        kernel_stats_rows = [["MM", "PGD", "RSS", "TOTAL_VM"], ["0", "0", "0k", "0k"]]
+        print_table(kernel_stats_rows)
+        print()
+        return
+
+    # Show memory statistics
+    rss_info = task_rss(task)
+    total_vm_kb = mm.total_vm.value_() * prog["PAGE_SIZE"].value_() // 1024
+
+    stats_rows: List[List[CellFormat]] = [
+        [
+            CellFormat("MM", "^"),
+            CellFormat("PGD", "^"),
+            CellFormat("RSS", "^"),
+            CellFormat("TOTAL_VM", "^"),
+        ],
+        [
+            CellFormat(mm.value_(), "^x"),
+            CellFormat(mm.pgd.value_(), "^x"),
+            CellFormat(f"{rss_info.total}k", "^"),
+            CellFormat(f"{total_vm_kb}k", "^"),
+        ],
+    ]
+    print_table(stats_rows)
+    # Show detailed VMA info
+    rows = [
+        [
+            CellFormat("VMA", "^"),
+            CellFormat("START", "^"),
+            CellFormat("END", "^"),
+            CellFormat("FLAGS", "<"),
+            CellFormat("FILE", "<"),
+        ]
+    ]
+    for vma in for_each_vma(mm):
+        file_path = vma_name(vma, mm)
+        rows.append(
+            [
+                CellFormat(f"{vma.value_():x}", "^"),
+                CellFormat(f"{vma.vm_start.value_():x}", "^"),
+                CellFormat(f"{vma.vm_end.value_():x}", "^"),
+                CellFormat(f"{vma.vm_flags.value_():x}", "<"),
+                CellFormat(file_path, "<"),
+            ]
+        )
+
+    print_table(rows)
+
+
+@crash_command(
+    description="virtual memory",
+    long_description="""This command displays basic virtual memory information of a context,
+consisting of a pointer to its mm_struct and page directory, its RSS and
+total virtual memory size; and a list of pointers to each vm_area_struct,
+its starting and ending address, vm_flags value, and file pathname. If no
+arguments are entered, the current context is used.
+""",
+    arguments=(
+        argument(
+            "pids_or_tasks",
+            metavar="pid | task",
+            nargs="*",
+            type="pid_or_task",
+            help="one or more process PIDs or hexadecimal task_struct pointers",
+        ),
+        drgn_argument,
+    ),
+)
+def _crash_cmd_vm(
+    prog: Program, name: str, args: argparse.Namespace, **kwargs: Any
+) -> None:
+
+    if args.drgn:
+        _generate_drgn_code(prog, args)
+        return
+
+    # Non-drgn mode
+    tasks = _get_tasks_from_args(prog, args.pids_or_tasks)
+
+    for task in tasks:
+        if task:  # Add null check
+            _print_task_vm_info(prog, task)
+            print()
```

Comment:
`crash_get_context()` already does all of the work here, so I think you can drop `_get_tasks_from_args()` and simplify this to:

```suggestion
    if not args.pids_or_tasks:
        args.pids_or_tasks.append(None)

    for task_arg in args.pids_or_tasks:
        try:
            task = crash_get_context(prog, task_arg)
        except Exception as e:
            print("vm:", e)
        else:
            _print_task_vm_info(prog, task)
        print()
```